components/board/genericlinux/hw_pwm.go

```diff
@@ -141,23 +152,37 @@ func (pwm *pwmDevice) SetPwm(freqHz uint, dutyCycle float64) (err error) {
 	activeDurationNs := uint64(float64(periodNs) * dutyCycle)
 
 	// If we ever try setting the active duration higher than the period (or the period lower than
-	// the active duration), we will get an error. Try setting one, then the other, then the first
-	// one again. If the first of those three had errors by being too small/large, the last one
-	// should take care of it. So, we purposely ignore any errors on the first value we try to
-	// write.
-	if err := pwm.writeLine("period", periodNs); err != nil {
-		// This is okay; we'll change the active duration and then try setting the period again.
-		pwm.logger.Debugf("Ignoring trouble setting the period on HW PWM device %s line %d: %s",
-			pwm.chipPath, pwm.line, err)
+	// the active duration), we will get an error. So, make sure we never do that!
+
+	// The BeagleBone has a weird quirk where, if you don't change the period of active duration
+	// after enabling the PWM line, it just goes high and stays there, rather than blinking at the
+	// intended period. To avoid this, we first set the active duration to 0 and the period to 1 ns,
```

Comment:
"intended period" feels wrong to me. "Intended frequency" feels better, but we're not doing anything with the frequency directly. Maybe "intended rate"?

Reply:
Comment doesn't match code: we're not setting the period to 1 ns any more because the Jetson couldn't handle that. Rephrase this and the last sentence of the paragraph.